File: primerTrimestre/introduccion_programacion_python/3repetitivas_py/ej11rep.py
# Programa: ej11rep.py
# Propósito: Suponiendo que hemos introducido una cadena por teclado que representa una frase (palabras separadas por espacios),
# realiza un programa que cuente cuantas palabras tiene.
#
# Autor: Antonio Quesada
# Fecha: 30/10/2019.
#
# Variables a usar:
#   * cadena
#   * contar
#   * palabras

#Pedimos que introduzca una cadena.
cadena = str(input("Introduce una cadena(frase):"))

#Bucle for y proceso.
#Inicializamos el contador.
contar = 0
    
#Utilizamos la funcion len para que nos devuelva la longitud de esa cadena de caracteres.
for i in range(1, len(cadena)):
    
  #Vamos recorriendo posiciones en la cadena para que lo cuente.   
  if cadena[i] == ' ' and cadena[i-1] != ' ':
    contar+=1
    palabras = contar + 1
    print ("La frase tiene", palabras, "palabras")
    
    
    
    

    

File: primerTrimestre/java basico/3repetitivas/Ej11rep.java
/**Suponiendo que hemos introducido una cadena por teclado que representa una frase (palabras separadas por espacios), 
 * realiza un programa que cuente cuantas palabras tiene.
 *
 * @author Antonio Quesada
 */

import java.util.Scanner;

public class Ej11rep {
  public static void main(String[] args) {
    Scanner s = new Scanner(System.in);
    String cadena;
    
    //Pedimos una cadena y un caracter.
    System.out.println("Introduce una cadena: ");
    cadena = s.nextLine();
    
    //Inicializamos contador a 0.
    int contador = 0;
    for (int i = 0; i < cadena.length(); i++){
      if (cadena.charAt(i)== ' '){
        contador++; //Contamos el número de espacios entre palabra y palabra.  
    }

  }
    int palabras;
    palabras = contador+1; //Contamos las palabras.
    System.out.println("La cadena tiene "+palabras+" palabras."); //Imprimimos el numero de palabras.
  }
}
